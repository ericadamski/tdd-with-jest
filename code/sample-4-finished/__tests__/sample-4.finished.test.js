const multiply = require("../sample-4.finished");

describe("Multiply", () => {
  it("should multiply two numbers", () => {
    // Arrange

    // Act
    const result = multiply(1, 2);

    // Assert
    expect(result).toBe(2);
  });

  it("should throw an error when attempting to multiply non-numbers", () => {
    // Arrange

    // Act
    const boolResult = () => multiply(true, false);
    const stringResult = () => multiply("hello", "world");
    const arrayResult = () => multiply([0], [1, 2]);
    const objectResult = () => multiply({ value: 1 }, 1);
    const undefinedResult = () => multiply(undefined, 1);
    const nullResult = () => multiply(null, 1);
    const functionResult = () =>
      multiply(function() {
        return 1;
      }, 1);

    // Assert
    expect(boolResult).toThrowErrorMatchingSnapshot();
    expect(stringResult).toThrowErrorMatchingSnapshot();
    expect(arrayResult).toThrowErrorMatchingSnapshot();
    expect(objectResult).toThrowErrorMatchingSnapshot();
    expect(undefinedResult).toThrowErrorMatchingSnapshot();
    expect(nullResult).toThrowErrorMatchingSnapshot();
    expect(functionResult).toThrowErrorMatchingSnapshot();
  });
});
