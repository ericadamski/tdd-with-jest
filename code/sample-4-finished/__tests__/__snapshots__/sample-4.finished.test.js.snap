// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiply should throw an error when attempting to multiply non-numbers 1`] = `"This isn't a number!"`;

exports[`Multiply should throw an error when attempting to multiply non-numbers 2`] = `"This isn't a number!"`;

exports[`Multiply should throw an error when attempting to multiply non-numbers 3`] = `"This isn't a number!"`;

exports[`Multiply should throw an error when attempting to multiply non-numbers 4`] = `"This isn't a number!"`;

exports[`Multiply should throw an error when attempting to multiply non-numbers 5`] = `"This isn't a number!"`;

exports[`Multiply should throw an error when attempting to multiply non-numbers 6`] = `"This isn't a number!"`;

exports[`Multiply should throw an error when attempting to multiply non-numbers 7`] = `"This isn't a number!"`;
